src/common/Utilities/Util.h

```diff
@@ -304,7 +304,8 @@ TC_COMMON_API bool IsIPAddress(char const* ipaddress);
 TC_COMMON_API uint32 CreatePIDFile(std::string const& filename);
 TC_COMMON_API uint32 GetPID();
 
-TC_COMMON_API std::string ByteArrayToHexStr(uint8 const* bytes, uint32 length, bool reverse = false);
+TC_COMMON_API std::string ByteArrayToHexStr(uint8 const* bytes, size_t length, bool reverse = false);
+template <typename C> std::string ByteArrayToHexStr(C const& container, bool reverse = false) { return ByteArrayToHexStr(std::data(container), std::size(container), reverse); }
```

Comment:
TC_COMMON_API lacking here kind of defeats the purpose of this method being here (and you can't add it)

Reply:
not sure i follow, `TC_*_API` shouldn't be necessary for any method defined in the header

Reply:
IIRC `TC_*_API` macros are for hotswap
except you can't have declspec(dllexport) (which is what those macros do) on template methods for obvious reasons

so you broke hotswap builds

Reply:
actually i don't know i just know TC_API macro breaks it here if its added and apparently it builds on nix for obscure reasons so ... ???

Reply:
`TC_COMMON_API` resolves to `export` inside common, and to `import` outside common

but that only matters if the definition is only in common, if the definition is in the header then it's in every compilation unit anyway